components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/util/ClaimUtil.java

```diff
@@ -83,13 +83,16 @@ private ClaimUtil() {
 
             UserStoreManager userstore = realm.getUserStoreManager();
 
-            TokenMgtDAO tokenMgtDAO = new TokenMgtDAO();
-            AccessTokenDO accessTokenDO = tokenMgtDAO.retrieveAccessToken(tokenResponse.getAuthorizationContextToken()
-                    .getTokenString(), false);
+            AccessTokenDO accessTokenDO = OAuth2Util.getAccessTokenDOfromTokenIdentifier(tokenResponse
+                    .getAuthorizationContextToken().getTokenString());
             ApplicationManagementService applicationMgtService = OAuth2ServiceComponentHolder.getApplicationMgtService();
-            String clientId = null;
+            String clientId;
             if (accessTokenDO != null) {
                 clientId = accessTokenDO.getConsumerKey();
+            } else {
+                // this means the token is not active so we can't proceed further
+                throw new UserInfoEndpointException(OAuthError.ResourceResponse.INVALID_TOKEN,
+                        "Invalid Access Token. Access token is not ACTIVE.");
```

Comment:
@Pushpalanka It is better to say what is the exact error in the exception. e.g.
"Could not find the AccessToken corresponding to the token string xyz. The token may be invalid or not ACTIVE"